src/ServiceProvider.php

```diff
@@ -58,17 +58,12 @@ public function register()
 	 */
 	public function boot()
 	{
-		if (version_compare(Application::VERSION, '5.3', '>=')) {
-			$this->loadMigrationsFrom(__DIR__.'/migrations');
-			$this->publishes([
-				__DIR__.'/config/config.php' => config_path('settings.php')
-			], 'config');
-		} else if (version_compare(Application::VERSION, '5.0', '>=')) {
+		if (version_compare(Application::VERSION, '5.0', '>=')) {
 			$this->publishes([
 				__DIR__.'/config/config.php' => config_path('settings.php')
 			], 'config');
 			$this->publishes([
-				__DIR__.'/migrations' => database_path('migrations')
+				__DIR__.'/migrations/2015_08_25_172600_create_settings_table.php' => database_path('migrations/'.date('Y_m_d_His').'_create_settings_table.php')
```

Comment:
Just curious, why would we add fresh timestamp? The idea is the timestamp represents a version.

Reply:
@bweston92 I think it's good to keep the order of the migrations files in an application correspond the order in which the migrations were run. If you leave the settings migration date, you create a migration backdating.

I don't know whether it is OK to backdate migrations in Laravel. If it is normal, I agree to leave the default timestamp. Anyway, I will be able to rename the migration as I want in the `database/migrations` directory.

Reply:
Ok but then on subsequent migrations you will have the same migration with a different file name.

Also I believe (not used Laravel in a while) it outputs the name of the files it has published.

It is absolutely ok to backdate migrations, migration file names are just checked against a database table.

I go back to still using the date as a version number @anlutro can judge this one.

Reply:
@bweston92 Ok, waiting for your decision.

Reply:
@FinesseRus I see that you've actually added "--tags" which would means the user would have to deliberately publish the migrations again.

- What happens if they don't pass tags will it generate the migration again with a different name?
- What version of Laravel did `--tags` get implemented?

Reply:
@bweston92 

> What happens if they don't pass tags will it generate the migration again with a different name?
Yes.

> What version of Laravel did --tags get implemented?
5.0.

I took the `--tag` option from the original readme:

> Publish the config file by running php artisan vendor:publish --provider="anlutro\LaravelSettings\ServiceProvider" --tag="config". The config file will give you control over which storage engine to use as well as some storage-specific settings.

If a user runs the commands as they are listed in the readme, he won't have excess migrations files (he will copy the config file and then copy the migration file if he decides to use a database storage).